### ISSUE TITLE
Extends ``use-set-for-membership`` for all hashable classes

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [PyCQA/pylint#4933]
### Current problem

In the following code ``use-set-for-membership`` is not raised for hashable classes.
```python
class Hashable:
    pass


class NotHashable:
    def __eq__(self, other):
        return self == other


class HashableAgain:
    def __eq__(self, other):
        return self == other

    def __hash__(self):
        return id(self)

x in (1, Hashable, HashableAgain)  # [use-set-for-membership]
x in (1, NotHashable)  # List is not hashable
```

### Desired solution

Raise  ``use-set-for-membership`` for both constants and hashable object.

### Additional context

https://github.com/PyCQA/pylint/pull/4841#issuecomment-908386541

**Comment by Pierre-Sassoulas** [comment]
In #7036 @jacobtylerwalls implemented a function that could be very handy for this:

https://github.com/PyCQA/pylint/blob/8033706deeedae809694bedf23fe6bb6798db980/pylint/checkers/utils.py#L1859
